app/webhooks/stripe/route.ts

```diff
@@ -82,33 +83,20 @@ export async function POST(req: Request) {
 					);
 				}
 				await handleSubscriptionCancellation(event.data.object);
+				await upsertSubscription(event.data.object.id);
 				break;
 
-			case "invoice.created":
-				console.log(`🔔  Invoice created: ${event.data.object.id}`);
+			case "invoice.created": {
+				console.log(`🔔 Invoice created: ${event.data.object.id}`);
 
-				// TODO: Skip for now - will be handled when implementing subscription cancellation invoice processing
-				if (
-					event.data.object.subscription &&
-					typeof event.data.object.subscription === "string"
-				) {
-					const subscriptionId = event.data.object.subscription;
-					const subscription =
-						await stripe.subscriptions.retrieve(subscriptionId);
-
-					if (subscription.status === "canceled") {
-						console.log(
-							"Skipping processing for canceled subscription invoice: ",
-							subscriptionId,
-						);
-						break;
-					}
-				}
+				await handleInvoiceCreation(event.data.object);
 
+				// TODO: This block will be removed in the other issue.
 				if (event.data.object.billing_reason === "subscription_cycle") {
 					await handleSubscriptionCycleInvoice(event.data.object);
 				}
```

Comment:
:memo: This block will be removed in the other issue. We plan to add user seats in other ways.